> 🤖 **Automated review** — generated by Claude Code on behalf of @metux. Not a human review.

## Update — `xserver-build-solaris` first run **failed** (1m46s)

**Verdict: 🔴 changes requested** — not a defect in the workflow wiring (that part is correct), but the new lane is red and the cause is structural, so it needs one fix before it can go green. Expected bring-up iteration.

### Root cause (job [83433602610](https://github.com/X11Libre/xserver/actions/runs/28170613470/job/83433602610))
The VM boots fine, but the build dies immediately:

```
./.github/scripts/solaris/run-xserver-build.sh[20]: meson: not found [No such file or directory]
##[error]ssh exited with code 127
```

`meson` is missing because **nothing got installed**. Timeline in the log: catalog refresh finishes at `12:40:07`, "meson not found" is at `12:40:07` — ~4 s later. Installing gcc14 + meson + ninja + all the X libs would take *minutes*; 4 s means `pkg install` aborted instantly and the failure was swallowed.

Two things combine:
1. **`install-pkg.sh` runs one big `pkg install A B C … || true`.** `pkg` is *transactional* — a single unknown/ambiguous FMRI aborts the **entire** set, so *zero* packages install. The trailing `|| true` then hides that, and the build script marches on without a toolchain.
2. **The `ooce/` publisher prefix is missing.** OmniOS-extra (publisher `extra.omnios`) FMRIs are stemmed under `ooce/` — e.g. `ooce/developer/build/meson`, `ooce/developer/gcc14`, `ooce/developer/build/ninja` — not the bare `developer/build/meson` the script uses. (At least one of the listed FMRIs is unresolvable, which is enough to abort the whole transaction.)

### Recommended fix
Split the install so the **build-critical toolchain is its own, fatal step** (no `|| true`), and let the optional X libs stay best-effort:

```sh
# toolchain — must succeed, and must be diagnosable if it doesn't
pkg install -v --accept \
    ooce/developer/gcc14 \
    ooce/developer/build/meson \
    ooce/developer/build/ninja \
    ooce/developer/build/pkg-config \
    ooce/developer/build/gnu-make \
    developer/versioning/git \
    runtime/python-312

# X libraries — best-effort during bring-up; meson will report any real gap
pkg install -v --accept \
    library/pixman x11/header/x11-protocols x11/library/libxau … || true
```

That way the next run either goes green or prints the *exact* bad FMRI by name (instead of the current silent "meson: not found"). The `ooce/` prefix above is the high-confidence guess; the X-lib FMRIs still need confirming against the r151058 catalog.

### Unchanged from the static review
Workflow wiring, YAML, `vmactions/omnios-vm@v1`, `-Dwerror=false` bring-up posture — all fine. No ABI/source impact; not a backport candidate.

Happy to push this `install-pkg.sh` restructuring and iterate on CI if you want — say the word.

--- a/.github/scripts/solaris/install-pkg.sh
+++ b/.github/scripts/solaris/install-pkg.sh
@@ -1,0 +1,68 @@
+#!/bin/sh
+#
+# Install xserver build dependencies on OpenIndiana (illumos), via IPS (`pkg`).
+#
+# OpenIndiana (unlike server-only OmniOS) is a desktop-capable illumos distro
+# and packages the full X stack — every X library, the protocol headers, Mesa
+# and the xcb-util family — under the single `openindiana.org` publisher in the
+# standard /usr layout. So everything is installable; nothing has to be built
+# from source here, and GLX can stay enabled (Mesa is packaged).
+#
+# FMRIs verified against the OpenIndiana Hipster catalog.
+#
+
+set -ex
+
+PATH="/usr/bin:/usr/sbin:/sbin:$PATH"
+export PATH
+
+pkg refresh || true
+
+# Build toolchain — MUST succeed. `pkg install` is transactional, so a single
+# unknown/ambiguous FMRI aborts the whole set; keeping this its own step
+# *without* `|| true` makes a bad FMRI fail loudly (and named) instead of
+# silently leaving us without a compiler/meson.
+pkg install -v --accept \
+    developer/gcc-14 \
+    developer/build/meson \
+    developer/build/ninja \
+    developer/build/pkg-config
+
+# Runtime + X libraries the server links against. Best-effort, one at a time, so
+# a single missing/misnamed FMRI only skips that package (and is named in the
+# log) rather than aborting the batch; meson will flag anything genuinely
+# required and absent.
+for pkg_fmri in \
+    runtime/python-313 \
+    library/zlib \
+    library/graphics/pixman \
+    system/library/freetype-2 \
+    x11/header/x11-protocols \
+    x11/header/xcb-protocols \
+    x11/font-util \
+    x11/library/libxau \
+    x11/library/libxcb \
+    x11/library/libxdmcp \
+    x11/library/libxext \
+    x11/library/libxfixes \
+    x11/library/libxrender \
+    x11/library/libxt \
+    x11/library/libx11 \
+    x11/library/xtrans \
+    x11/library/libxkbfile \
+    x11/library/libxfont2 \
+    x11/library/libfontenc \
+    x11/library/libxcvt \
+    x11/library/libxshmfence \
+    x11/library/libepoxy \
+    x11/library/xcb-util \
+    x11/library/xcb-util-wm \
+    x11/library/mesa
+do
+    # rc 0 = installed, rc 4 = already present / nothing to do — both fine.
+    rc=0
+    pkg install -v --accept "$pkg_fmri" || rc=$?
+    if [ "$rc" -ne 0 ] && [ "$rc" -ne 4 ]; then
+        echo "WARN: 'pkg install $pkg_fmri' failed (rc=$rc) — continuing (meson will flag it if required)"
+    fi
+done

--- a/.github/scripts/solaris/run-xserver-build.sh
+++ b/.github/scripts/solaris/run-xserver-build.sh
@@ -1,0 +1,32 @@
+#!/bin/sh
+#
+# Build the X server inside an OpenIndiana (illumos) VM. Mirrors the BSD
+# run-xserver-build.sh scripts: install deps, then meson setup/compile/install.
+#
+
+set -e
+
+./.github/scripts/solaris/install-pkg.sh
+
+echo "--> running xserver build ...."
+
+PATH="/usr/bin:/usr/sbin:/sbin:$PATH"
+export PATH
+
+# OpenIndiana installs into the standard /usr layout; build 64-bit (the illumos
+# gcc still defaults to 32-bit) and point pkg-config at the amd64 .pc dir plus
+# the noarch share/pkgconfig (xorgproto protocol files).
+CFLAGS="-m64 ${CFLAGS:-}"
+LDFLAGS="-m64 ${LDFLAGS:-}"
+PKG_CONFIG_PATH="/usr/lib/amd64/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:${PKG_CONFIG_PATH:-}"
+export CFLAGS LDFLAGS PKG_CONFIG_PATH
+
+export MESON_BUILDDIR=_build
+
+rm -rf "$MESON_BUILDDIR"
+meson setup "$MESON_BUILDDIR" $MESON_ARGS
+meson configure "$MESON_BUILDDIR"
+meson compile -v -C "$MESON_BUILDDIR" $jobcount $ninja_args
+# tests not wired up for this platform yet
+# meson test -C "$MESON_BUILDDIR" --print-errorlogs $MESON_TEST_ARGS
+meson install --no-rebuild -C "$MESON_BUILDDIR" $MESON_INSTALL_ARGS

--- a/.github/workflows/build-xserver.yml
+++ b/.github/workflows/build-xserver.yml
@@ -422,6 +422,32 @@ jobs:
                   release: '10.1'
                   run:     ./.github/scripts/netbsd/run-xserver-build.sh
 
+    # illumos / Solaris build, run in an OpenIndiana VM (vmactions, same
+    # mechanism as the BSD jobs above). Exercises the Solaris-specific os/ code
+    # paths (procfs/psinfo, the sunos meson branch) that no other CI covers.
+    # OpenIndiana packages the full X stack incl. Mesa, so GLX stays enabled.
+    # werror is off for now since this platform is freshly added; tighten once
+    # it is green.
+    xserver-build-solaris:
+        runs-on: ubuntu-latest
+        needs: ubuntu-fetch-pkg
+        env:
+            # dri2/dri3 disabled: illumos has no kernel DRM/KMS, so the
+            # libdrm-backed direct-rendering extensions can't work (and OI's
+            # xf86drm.h pulls a drm.h that only exists for Linux). glx stays on
+            # (Mesa is packaged) for indirect/software GLX.
+            MESON_ARGS: -Dwerror=false -Dxvfb=true -Dxnest=true -Dxorg=false -Dxephyr=false -Dxfbdev=false -Ddri2=false -Ddri3=false
+        steps:
+            - uses: actions/checkout@v6
+            - uses: ./.github/actions/ubuntu-pkg-cache
+            - name: run in OpenIndiana (illumos) VM
+              id: vm
+              uses: vmactions/openindiana-vm@v1
+              with:
+                  envs: 'MESON_ARGS'
+                  usesh: true
+                  run:     ./.github/scripts/solaris/run-xserver-build.sh
+
     xserver-build-alpine:
         runs-on: ubuntu-latest
         needs: ubuntu-fetch-pkg

